script.js

```diff
@@ -1,4 +1,419 @@
+
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    console.log("Current Suit : " + currentSuit);
+
+    // rank 1-13
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suits: currentSuit,
+        rank: rankCounter,
+      };
+      console.log("rank : " + rankCounter);
+
+      cardDeck.push(card);
+      rankCounter = rankCounter + 1;
+    }
+    suitIndex = suitIndex + 1;
+  }
+  return cardDeck;
+};
+
+// generate the function from above into variable so easy to use
+var cardDeck = makeDeck();
+
+// helps to get all the random numbers
+var getRandomNumberIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// pulls the random number as an randomindex and search into the card deck for the number
+var shuffleCards = function (cardDeck) {
+  currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomNumberIndex(cardDeck.length);
+    var randomCardGenerated = cardDeck[randomIndex]; // this takes the random index which is a random number and then it will look for that number on the cards deck.
+    var currentCardGenerated = cardDeck[currentIndex]; // using the current index it will find a card in the deck
+    // it swaps random card and current card together for a result
+    cardDeck[currentIndex] = randomCardGenerated;
+    cardDeck[randomIndex] = currentCardGenerated;
+
+    currentIndex = currentIndex + 1;
+  }
+
+  return cardDeck;
+};
+
+// store the user name
+var username = [];
+var gameMode = "enter name";
+
+var checkUsername = function (input) {
+  gameMode == "enter name";
+  if (input != ``) {
+    gameMode = "start game";
+
+    username = input;
+    console.log("Check username input:", username);
+    var message = "Hi 😀" + username + " <br> </br> Click submit to play.🃏";
+  } else if (input == ``) {
+    gameMode == "enter name";
+    message = "Please enter your name 👀.";
+    return message;
+  }
+  return message;
+};
+
+// store the player scores in global variable from functions below
+var playerOneTwoCardResult = [];
+var computerScore = [];
+// shows the the player what card has been drawn
+var shuffledCardResultScreen = function () {
+  var firstCardValue = [];
+  console.log("check first card value global variable", firstCardValue);
+
+  var dealerFirstCardValue = [];
+  console.log("check global variable", dealerFirstCardValue);
+
+  var cardShuffled = shuffleCards(cardDeck);
+  console.log("card shuffed part 1", cardShuffled);
+  gameMode = "stage 2";
+
+  // player's first card
+  var playerOneCardOne = cardShuffled.pop();
```

Comment:
consider creating a function that can be used for generating both the player and computer hands and storing the cards in an array so that the individual cards do not have to be named , eg,` var playerOneCardOne etc`